src/components/Header/Header.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { TodoContext } from '../../TodoContext';
+import classNames from 'classnames';
+
+type Props = {};
+
+export const Header: React.FC<Props> = () => {
+  const [title, setTitle] = useState('');
+  const { todos, setTodos, addTodo } = useContext(TodoContext);
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const allCompleted = completedTodos.length === todos.length;
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const focusInput = () => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  };
+
+  const handleCompleteAll = () => {
+    if (allCompleted) {
+      setTodos(todos.map(todo => ({ ...todo, completed: false })));
+
+      return;
+    }
+
+    setTodos(todos.map(todo => ({ ...todo, completed: true })));
+  };
+
+  useEffect(() => {
+    focusInput();
+  }, [todos.length]);
+
+  return (
+    <header className="todoapp__header">
+      {todos.length > 0 && (
+        <button
+          type="button"
+          className={classNames('todoapp__toggle-all', {
+            active: allCompleted,
+          })}
+          data-cy="ToggleAllButton"
+          onClick={handleCompleteAll}
+        />
+      )}
+
+      <form
+        onSubmit={event => {
+          event.preventDefault();
+
+          if (title.trim() === '') {
+            return;
+          }
+
+          addTodo(title);
+          setTitle('');
+        }}
```

Comment:
Pls move the function inside `onSubmit` handler to a separate const and use as a callback here